routing/notifications.go

```diff
@@ -117,8 +117,9 @@ type topologyClient struct {
 // notifyTopologyChange notifies all registered clients of a new change in
 // graph topology in a non-blocking.
 func (r *ChannelRouter) notifyTopologyChange(topologyDiff *TopologyChange) {
-	r.RLock()
-	defer r.RUnlock()
+	// Get the clients inside a lock.
+	r.topologyClientMutex.RLock()
```

Comment:
Maybe I'm missing something but it seems to me that the current iteration of the PR just renames the lock?

Reply:
I think it decreases the lock scope to only locking the map here. Suppose we have a different field `foo` we wanna lock in `r` we can then add an `r.fooMutex`. So instead of locking the whole struct we only lock a specific mutex. Does it make sense?

Reply:
AFAIK it's just a syntax thing and the mutex still locks scope where it's called. So embedding the mutex into the struct doesn't guarantee automatic locking of all members.

Reply:
Yeah I guess you can call it a syntax change. The idea was to have one dedicated lock variable for this map, so we can use more locks in needed. Maybe better explained in [this example](https://go.dev/play/p/EQMzJ13MzV5). However I now think we can easily accomplish it using our own [sync maps](https://github.com/lightningnetwork/lnd/blob/master/lntemp/node/sync_map.go), as it's optimized for "when the entry for a given key is only ever written once but read many times", fits our case here well.

Reply:
Would be interesting to see the change's performance impact to evaluate that the `SyncMap` is indeed faster vs the rw lock.

Reply:
Yeah, cool, thanks for the test! I think it's probably a good idea to use the `SyncMap` then.